apps/cloudbroker/cloudbrokerlib/netmgr.py

```diff
@@ -337,12 +337,31 @@ def fw_list(self, gid, domain=None, **kwargs):
             result.append(vfwdict)
         return result
 
-    def fw_start(self, fwid, **kwargs):
+    def fw_start(self, fwid, resettype='restore', targetNid=None, **kwargs):
         """
         param:fwid firewall id
         param:gid grid id
         """
+        if resettype not in ['factory', 'restore']:
+            raise exceptions.BadRequest("Invalid value {} for resettype".format(resettype))
         fwobj = self._getVFWObject(fwid)
+        self.fw_delete(fwid, fwid.gid)
+        if resettype == 'restore':
+            restored = self.fw_restore(fwid, targetNid)
+        if resettype == 'factory' or not restored:
+            cloudspace = self.cbmodel.cloudspace.get(int(fwobj.domain))
+            pool = self.cbmodel.externalnetwork.get(cloudspace.externalnetworkId)
+            if cloudspace.externalnetworkip is None:
+                raise exceptions.BadRequest('Can not reset VFW which has no external network IP please deploy instead.')
```

Comment:
you probably want to write this test in the beginning before doing anything else